src/test/esperanza/finalizationstate_withdraw_tests.cpp

```diff
@@ -28,37 +28,6 @@ BOOST_AUTO_TEST_CASE(validate_withdraw_not_a_validator) {
                     +Result::WITHDRAW_NOT_A_VALIDATOR);
 }
 
-BOOST_AUTO_TEST_CASE(process_withdraw_before_end_dynasty) {
-
-  FinalizationStateSpy spy;
-  CAmount withdrawAmount = 0;
-  uint160 validatorAddress = RandValidatorAddr();
-  CAmount depositSize = spy.MinDepositSize();
-
-  // For simplicity we keep the targetHash constant since it does not
-  // affect the state.
-  uint256 targetHash = GetRandHash();
-  CBlockIndex block_index;
-  block_index.phashBlock = &targetHash;
-  spy.SetRecommendedTarget(block_index);
-
-  spy.CreateAndActivateDeposit(validatorAddress, depositSize);
-
-  BOOST_CHECK_EQUAL(spy.ValidateLogout(validatorAddress), +Result::SUCCESS);
-  spy.ProcessLogout(validatorAddress);
-
-  for (uint32_t i = 6; i < spy.DynastyLogoutDelay() + 1; ++i) {
-    Vote vote{validatorAddress, targetHash, i - 2, i - 1};
-    BOOST_CHECK_EQUAL(spy.ValidateVote(vote), +Result::SUCCESS);
-    spy.ProcessVote(vote);
-
-    BOOST_CHECK_EQUAL(spy.InitializeEpoch(1 + i * spy.EpochLength()), +Result::SUCCESS);
-  }
-
-  BOOST_CHECK_EQUAL(spy.ValidateWithdraw(validatorAddress, withdrawAmount),
-                    +Result::WITHDRAW_BEFORE_END_DYNASTY);
-}
```

Comment:
Why this test has been removed?

Reply:
We removed "unused" `WITHDRAW_BEFORE_END_DYNASTY` state. 
But even if we had it, you can see in master that the next test includes this one so was duplicated/executed twice.

Reply:
Resolving. Feel free to re-open if you have suggestions for improvements